feat: generalize test container runner to support podman

derekthecool opened the pull request:
(original pr committed through `gh using the gh pr create --fill` which did not create a body here)

This code change adds support for the tests to check through docker and podman to see which container runner to use.
In every case I've found the statement of podman being a drop in replacement for docker certainly holds, this case included.

Then I replaced all calls to docker with this

```ps1
& $ContainerRunner ....
```

derekthecool committed:
feat: generalize test container runner to support podman

fsackur commented on tests/ProfileAsync.Tests.ps1:
```diff
@@ -2,6 +2,7 @@ BeforeAll {
     $TestRoot = $PSScriptRoot
     $ModuleBase = $PSScriptRoot | Split-Path
     $PesterBase = (Get-Module Pester).ModuleBase
+    $ContainerRunner = 'docker', 'podman' | ForEach-Object { Get-Command $_ -ErrorAction SilentlyContinue } | Select-Object -Expand Name
```
Fails for me, with both installed, with `CommandNotFoundException: The term 'docker podman' is not recognized...`

Could you add `-First 1` to the Select-Object, to fix that? I think docker should take precedence when both are installed.

fsackur reviewed (changes requested):
Thanks for the PR!

`podman` didn't complete the test on my dev machine, does it work for you?

If you say it works, and if docker takes precedence when both are installed, then I'm happy to merge. Please fix the review comment, though!

derekthecool committed:
Fix docker/podman check to take the first found runner

This may benefit from some error checking if no runners are found.
But this should at least take care of cases when both podman and docker are installed.

derekthecool commented:
Oops, my bad. I didn't have docker installed and forgot about what might happen when both exist.
I added the `-First 1`, that should certainly take care of the problem.
